### PR TITLE
Quick instructions to compile on Fedora 37

### PR DESCRIPTION
**Pull request by MagnoBooter**
This PR needs some verification and testing on a second F37 install before merge.

### CODE REVIEW

**Review by umireon — CHANGES_REQUESTED** [review]
Please rename the filename to `BUILDING-FEDORA.md`.

File: docs/BUILDING-FEDORA-37.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+_As a privileged user (root):_  
+You then need to copy the plugin binary to `/usr/lib64/obs-plugins`:
+```
+# cp build_x86_64/obs-backgroundremoval.so /usr/lib64/obs-plugins/
+```
+
+And copy the contents of the [data](../data) directory to `/usr/share/obs/obs-plugins/obs-backgroundremoval`:
+```
+# cp -r data /usr/share/obs/obs-plugins/obs-backgroundremoval
+```
```
Comment (umireon) [inline docs/BUILDING-FEDORA-37.md]: Please fix the contents as follows:

## Instructions for building the plugin on Fedora

First, you have to install the development tools:

```
sudo dnf groupinstall "Development Tools"
```

Then, make sure you have the dependencies of this plugin installed:

```
sudo dnf install cmake gcc-c++ ninja-build obs-studio-devel qt6-qtbase-devel
```

Clone the repository and set up the submodules:

```
git clone https://github.com/royshil/obs-backgroundremoval.git
cd obs-backgroundremoval
git submodule update --init
```

After cloning, use the following command to compile:

```
.github/scripts/build-linux --skip-deps
```

You then need to install the plugin binary to `/usr`:

```
sudo cmake --install build_x86_64 --prefix /usr
```

**Review by umireon — CHANGES_REQUESTED** [review]

File: docs/BUILDING-FEDORA.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+First, you have to install the development tools:
+```
+sudo dnf groupinstall "Development Tools"
+```
+
+Then, make sure you have the dependencies of this plugin installed:
+
+```
+sudo dnf install cmake gcc-c++ libxkbcommon-devel ninja-build obs-studio-devel opencv-devel qt6-qtbase-devel
```
Comment (umireon) [inline docs/BUILDING-FEDORA.md]: Do you really need to install libxkbcommon-devel?

File: docs/BUILDING-FEDORA.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+```
+git clone https://github.com/royshil/obs-backgroundremoval.git
+cd obs-backgroundremoval
+git submodule update --init
+```
+
+Before you compile, edit `cmake/linux/compilerconfig.cmake`, find this line: `set(_obs_gcc_c_options` and add the following line to the list of options:
+```
+    -fPIC
+```
```
Comment (umireon) [inline docs/BUILDING-FEDORA.md]: This will be fixed by #398 so we don't need to include this workaround in this document.

**Review by umireon — APPROVED** [review]
LGTM

**Comment by umireon** [comment]
@royshil I'm OK with merging this. This document will help Fedora people.

File: docs/BUILDING-FEDORA.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+## Instructions for building the plugin on Fedora 37
```
Comment (umireon) [inline docs/BUILDING-FEDORA.md]:
```suggestion
## Instructions for building the plugin on Fedora
```

File: docs/BUILDING-FEDORA.md (hunk 2 above)
Comment (MagnoBooter) [reply docs/BUILDING-FEDORA.md]: Well, maybe not? :smile:  But I had an error message that looked like this: 
```
[...]
-- Found OpenGL: /usr/lib64/libOpenGL.so   
-- Found WrapOpenGL: TRUE  
-- Could NOT find XKB (missing: XKB_LIBRARY XKB_INCLUDE_DIR) (Required is at least version "0.5.0")
-- Found WrapVulkanHeaders: /usr/include  
[...]
```
But it seems to compile anyway, so I'll remove the dependency.

File: docs/BUILDING-FEDORA.md (hunk 4 above)
Comment (MagnoBooter) [reply docs/BUILDING-FEDORA.md]: ok, will fix this in the next commit :+1:

**Review by royshil — APPROVED** [review]
yep - go ahead!

**Pull request merged**